src/test/java/org/beanpod/switchboard/controller/DecoderControllerTest.java

```diff
@@ -42,14 +49,19 @@
     static private DeviceEntity device1, device2;
     static private DecoderEntity decoder1, decoder2;
     static private List<DecoderEntity> listOfDecoders;
+    static private Set<ChannelEntity> setOfChannels;
 
     @BeforeAll
-    static void decoderFixture(){
+    static void decoderFixture() throws ParseException {
         //stubbing device and decoder objects
         device1 = DeviceFixture.getDevice1();
         device2 = DeviceFixture.getDevice2();
-        decoder1 = DecoderFixture.getDecoder1(device1);
-        decoder2 = DecoderFixture.getDecoder2(device2);
+
+        setOfChannels = ChannelFixture.getSetOfChannels();
+
+        decoder1 = DecoderFixture.getDecoder1(device1,setOfChannels);
+        decoder2 = DecoderFixture.getDecoder2(device2,setOfChannels);
```

Comment:
I think our fixtures should access ChannelFixture on their own to get the set of channels. That way you don't have to build anything in the test code.

Reply:
fixed